starlark/library.go

```diff
@@ -888,6 +889,32 @@ func print(thread *Thread, b *Builtin, args Tuple, kwargs []Tuple) (Value, error
 	return None, nil
 }
 
+// TODO: add link to spec.md
+func fail(thread *Thread, b *Builtin, args Tuple, kwargs []Tuple) (Value, error) {
+	if len(kwargs) > 0 {
+		return nil, fmt.Errorf("fail does not accept keyword arguments")
+	}
+	if len(args) == 0 {
+		return None, fmt.Errorf("none")
+	}
+	attr := ""
+	if err := UnpackPositionalArgs("fail", args, kwargs, 0, &args[0], &attr); err != nil {
+		return nil, err
+	}
+	buf := new(strings.Builder)
+	if len(attr) > 0 {
+		buf.WriteString(fmt.Sprintf("attribute %s: ", attr))
```

Comment:
fmt.Fprintf(buf, ...)

buf implements io.Writer, so you can Fprintf directly to it.